src/types/mod.rs

```diff
@@ -522,6 +522,17 @@ pub struct ObjectCandidate {
     pub debug: ObjectUseInfo,
 }
 
+/// A URI representing an [`ObjectFileSource`].
+///
+/// Note that this does not provide enough information to download the object file, for this
+/// you need the actual [`ObjectFileSource`].  The purpose of this URI is to be able to
+/// display to a user who might be able to use this in other tools.  E.g. for an S3 source
+/// this could be an `s3://` URI etc.
+///
+/// [`ObjectFileSource`]: crate::services::download::ObjectFileSource
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
+pub struct ObjectFileSourceURI(String);
```

Comment:
_nit_: Would `ObjectUri` suffice as a shorter name?

Reply:
meh, doesn't really matter but mostly i think that's too short for such an auxiliary type